### DISCUSSION TITLE
Обсуждение новой функциональности ЯМузыки: "Моя волна"

### DISCUSSION THREAD

**Discussion by KisoidPoh095** [Chimildic/YaMuTools#30] [Беседка]
Сегодня заметил, что на странице https://music.yandex.ru/home появилась новая функция. Оформлена красиво: прямо в центре крутятся разноцветные "волны". Не заметить невозможно. Цветовая гамма и скорость кручения, по видимости, зависят от "настроения".
**Новости по теме:**
[1] https://appleinsider.ru/obzory-prilozhenij/v-yandeks-muzyke-poyavilas-funkciya-kotoraya-vam-tochno-ponravitsya.html
[2] https://dtf.ru/music/943516-v-yandeks-muzyke-poyavilas-moya-volna-eto-beskonechnyy-personalnyy-potok-trekov-s-uchetom-bolee-tysyachi-faktorov

Насколько я понял, это что-то типа смеси Радио, Дежавю и Тайника. Только более интерактивной: по идее, нажатие/ненажатие сердечка, пропуск треков и т.д. должны в реалтайме влиять на состав подборки.

В [1] заявлено, что новый рекомендательный алгоритм спасает от "замыкания в «пузыре рекомендаций»".
Вот уж не знаю... Как по мне, "информационный пузырь" aka "пузырь фильтров" - это фундаментальная проблема любых рекомендательных алгоритмов, основанных на усреднении или ассоциативных связях. Будь то нейросети, формула Байеса или минимизация квадратичных отклонений... Ну посмотрим.

Также надеюсь, что тут не случится как с Яндекс.Дзеном. Это когда в рекламе говорят, что "Дзен вызывает появление новых интересов", а по факту, начинаешь всё дизлайкать, и лента на 99% наполняется видео с котятами...

Или как в Кинопоиске. Сначала думаешь: "я такой хитрый, щас как поставлю кучу положительных оценок всяким необычным фильмам и заминусую нафиг всякую банальщину типа Побег из Шоушенка...". А в результате получаешь в качестве рекомендаций список непросмотренных фильмов из ТОП-500 Кинопоиска, только отсортированный в другом порядке...

**Comment by sergvpr** [comment]
Могу точно сказать, что новая волна полный отстой. Несколько раз писал в поддержку, чтобы поправили алгоритм, но воз там же.
Постоянные повторы треков, которые играли вчера, неделю назад (это раздражает). То есть история вообще никак не вычитается.
Это не просто ощущения, я проверяю в ЛастФМ, так как всё скроблится туда.
Перестал слушать Волну из-за этого.
Я знаю, что есть настройка Новое, но я хочу чтобы подмешивались мои треки, которые я давно не слушал.

**Reply by kisoid** [reply]
>Постоянные повторы треков, которые играли вчера, неделю назад

Есть такое. Я просто включил фильтр "только новое". В качестве ещё одного источника нового она очень даже ничего.

Если бы была настройка "Проигрывать трек не чаще, чем 1 раз в N прослушиваний"... Но навряд ли они будут заморачиваться с этим. К тому же у них проблема с определением одинаковых треков. Это уже вопрос к музыкальной редакции - как они заводят альбомы, проставляют теги и т.д.

Глобально для себя решил проблему с повторами так:
- делаю временную копию Мне нравится
- перемешиваю треки внутри копии случайным образом с помощью YaMuTools
- слушаю от начала до конца, каждый раз удаляя недавно прослушанные с помощью YaMuTools

Таким образом получается, что я равномерно прослушиваю весь плейлист Мне нравится без повторов. Параллельно закидываю в оригинальный "Мне нравится" новые интересные треки и удаляю надоевшие. На один цикл уходит 1-2 месяца (размер плейлиста чуть больше 2000 композиций).
